File: src/main/java/io/zeitwert/fm/building/model/ObjBuildingPartRating.java

package io.zeitwert.fm.building.model;

import java.time.LocalDate;
import java.util.List;

import io.dddrive.obj.model.ObjPart;
import io.zeitwert.fm.building.model.enums.CodeBuildingMaintenanceStrategy;
import io.zeitwert.fm.building.model.enums.CodeBuildingPart;
import io.zeitwert.fm.building.model.enums.CodeBuildingPartCatalog;
import io.zeitwert.fm.building.model.enums.CodeBuildingRatingStatus;
import io.zeitwert.fm.oe.model.ObjUserFM;

public interface ObjBuildingPartRating extends ObjPart<ObjBuilding> {

	CodeBuildingRatingStatus getRatingStatus();

	void setRatingStatus(CodeBuildingRatingStatus ratingStatus);

	LocalDate getRatingDate();

	Integer getRatingYear();

	void setRatingDate(LocalDate ratingDate);

	ObjUserFM getRatingUser();

	void setRatingUser(ObjUserFM ratingUser);

	CodeBuildingPartCatalog getPartCatalog();

	void setPartCatalog(CodeBuildingPartCatalog partCatalog);

	CodeBuildingMaintenanceStrategy getMaintenanceStrategy();

	void setMaintenanceStrategy(CodeBuildingMaintenanceStrategy strategy);

	Integer getElementCount();

	ObjBuildingPartElementRating getElement(Integer seqNr);

	List<ObjBuildingPartElementRating> getElementList();

	ObjBuildingPartElementRating getElementById(Integer elementId);

	ObjBuildingPartElementRating getElement(CodeBuildingPart buildingPart);

	void clearElementList();

	ObjBuildingPartElementRating addElement(CodeBuildingPart buildingPart);

	void removeElement(Integer elementId);

	Integer getElementWeights();

	Integer getCondition();

	Integer getCondition(Integer year);

}

File: fm-domain/src/main/java/io/zeitwert/fm/building/model/ObjBuildingPartRating.kt
package io.zeitwert.fm.building.model

import dddrive.app.obj.model.ObjPart
import dddrive.ddd.property.model.PartListProperty
import io.dddrive.oe.model.ObjUser
import io.zeitwert.fm.building.model.enums.CodeBuildingMaintenanceStrategy
import io.zeitwert.fm.building.model.enums.CodeBuildingPart
import io.zeitwert.fm.building.model.enums.CodeBuildingPartCatalog
import io.zeitwert.fm.building.model.enums.CodeBuildingRatingStatus
import java.time.LocalDate

interface ObjBuildingPartRating : ObjPart<ObjBuilding> {

	var ratingStatus: CodeBuildingRatingStatus?

	var ratingDate: LocalDate?

	val ratingYear: Int?

	var ratingUserId: Any?

	var ratingUser: ObjUser?

	var partCatalog: CodeBuildingPartCatalog?

	var maintenanceStrategy: CodeBuildingMaintenanceStrategy?

	val elementList: PartListProperty<ObjBuildingPartElementRating>

	fun getElement(buildingPart: CodeBuildingPart): ObjBuildingPartElementRating

	fun addElement(buildingPart: CodeBuildingPart): ObjBuildingPartElementRating

	val elementWeights: Int

	val condition: Int?

	fun getCondition(year: Int): Int?

}
